homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -181,6 +193,75 @@ class EnvoyConsumptionSensorEntityDescription(
 )
 
 
+@dataclass
+class EnvoyEnchargeRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[EnvoyEncharge], datetime.datetime | int | float]
+
+
+@dataclass
+class EnvoyEnchargeSensorEntityDescription(
+    SensorEntityDescription, EnvoyEnchargeRequiredKeysMixin
+):
+    """Describes an Envoy Encharge sensor entity."""
+
+
+@dataclass
+class EnvoyEnchargePowerRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[EnvoyEnchargePower], int | float]
+
+
+@dataclass
+class EnvoyEnchargePowerSensorEntityDescription(
+    SensorEntityDescription, EnvoyEnchargePowerRequiredKeysMixin
+):
+    """Describes an Envoy Encharge sensor entity."""
+
+
+ENCHARGE_INVENTORY_SENSORS = (
+    EnvoyEnchargeSensorEntityDescription(
+        key="temperature",
+        translation_key="current_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        value_fn=lambda encharge: encharge.temperature,
+    ),
+    EnvoyEnchargeSensorEntityDescription(
+        key=LAST_REPORTED_KEY,
+        translation_key=LAST_REPORTED_KEY,
+        native_unit_of_measurement=None,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda encharge: dt_util.utc_from_timestamp(encharge.last_report_date),
+    ),
+)
+ENCHARGE_POWER_SENSORS = (
+    EnvoyEnchargePowerSensorEntityDescription(
+        key="soc",
+        translation_key="state_of_charge",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        value_fn=lambda encharge: encharge.soc,
+    ),
+    EnvoyEnchargePowerSensorEntityDescription(
+        key="apparent_power_mva",
+        translation_key="apparent_power",
+        native_unit_of_measurement=UnitOfApparentPower.VOLT_AMPERE,
+        device_class=SensorDeviceClass.APPARENT_POWER,
+        value_fn=lambda encharge: encharge.apparent_power_mva * 0.001,
+    ),
+    EnvoyEnchargePowerSensorEntityDescription(
+        key="real_power_mw",
+        translation_key="real_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        value_fn=lambda encharge: encharge.real_power_mw * 0.001,
+    ),
+)
```

Comment:
Does anything here need one of the following?
- suggested_unit_of_measurement
- suggested_display_precision